src/NLog/Targets/EventLogTarget.cs

```diff
@@ -260,8 +265,17 @@ protected override void Write(LogEventInfo logEvent)
                 category = Convert.ToInt16(this.Category.Render(logEvent), CultureInfo.InvariantCulture);
             }
 
+            long maxKilobytes = 0;
+
+            if(this.MaxKilobytes != null)
+            {
+                maxKilobytes = Convert.ToInt64(this.MaxKilobytes.Render(logEvent), CultureInfo.InvariantCulture);
```

Comment:
maybe it's a bit odd to set this property, when writing every message?

First I was thinking to move it to `InitializeTarget`, but then you have a problem with `GetEventLog(logEvent)`.

Maybe we could cache the value? I was think to move it to `GetEventLog` and maybe creating a method for `isCacheUpToDate `